### PR TITLE
Add MockMvc HtmlUnit Support

### PR DESCRIPTION
**Pull request by rwinch**
# Important NOTE

I'm posting this for review. Before we merge this, we need to wait until HtmlUnit 2.18 is released to support httpcomponents 4.5. See [SPR-13158](https://jira.spring.io/browse/SPR-13158).

Issues [SPR-13158](https://jira.spring.io/browse/SPR-13158)

### CODE REVIEW

File: spring-test/src/main/java/org/springframework/test/web/servlet/htmlunit/webdriver/MockMvcHtmlUnitDriverBuilder.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.htmlunit.MockMvcWebConnectionBuilderSupport;
+import org.springframework.test.web.servlet.setup.MockMvcConfigurer;
+import org.springframework.web.context.WebApplicationContext;
+
+/**
+ * Convenience class for building an HtmlUnitDriver that will delegate to MockMvc and
+ * optionally delegate to an actual connection for specific requests.
+ *
+ * By default localhost will delegate to MockMvc and any other URL will delegate
```
Comment (sbrannen) [inline spring-test/src/main/java/org/springframework/test/web/servlet/htmlunit/webdriver/MockMvcHtmlUnitDriverBuilder.java]: and any other URL will delegate.... to what?

Comment (rwinch) [inline spring-test/src/main/java/org/springframework/test/web/servlet/htmlunit/webdriver/MockMvcHtmlUnitDriverBuilder.java]: to the original WebClient (i.e. actually perform the request)

Comment (sbrannen) [inline spring-test/src/main/java/org/springframework/test/web/servlet/htmlunit/webdriver/MockMvcHtmlUnitDriverBuilder.java]: Thanks. I've updated the documentation in the feature branch now (and _rewritten history_ again :wink:).

Comment (rwinch) [inline spring-test/src/main/java/org/springframework/test/web/servlet/htmlunit/webdriver/MockMvcHtmlUnitDriverBuilder.java]: Thank you!

**Comment by sbrannen** [comment]
Merged into `master` in 98c7a63bb78996d14e58f8db38fc3b8fdd3c7306.

**Pull request closed without merging**